cmd/insprd/memory/tree/dapp.go

```diff
@@ -247,122 +273,128 @@ func (amm *AppPermTreeGetter) Get(query string) (*meta.App, error) {
 
 }
 
-// ResolveBoundary is the recursive method that resolves connections for dApp boundaries
-// returns a map of boundary to  their respective resolved channel query
-func (amm *AppMemoryManager) ResolveBoundary(app *meta.App, usePermTree bool) (map[string]string, error) {
-	l := amm.logger.With(zap.String("operation", "boundary-resolution"), zap.String("dapp", app.Meta.Name))
-	l.Debug("received boundary resolution request", zap.Bool("useperm", usePermTree))
+func (amm *AppMemoryManager) ResolveBoundary(app *meta.App, usePermTree bool) (map[string]string, map[string]string, error) {
+	parent, _ := amm.GetParent(app, usePermTree)
 
-	boundaries := make(map[string]string)
-	unresolved := metautils.StrSet{}
-	for _, bound := range app.Spec.Boundary.Channels.Input.Union(app.Spec.Boundary.Channels.Output) {
-		boundaries[bound] = fmt.Sprintf("%s.%s", app.Meta.Name, bound)
-		unresolved[bound] = true
+	merr := ierrors.MultiError{
+		Errors: []error{},
 	}
 
-	var parentApp *meta.App
-
-	if usePermTree {
-		parApp, err := amm.Perm().Apps().Get(app.Meta.Parent)
+	resolvedRoutes := make(map[string]string)
+	boundaries := app.Spec.Boundary.Routes
+	for _, bound := range boundaries {
+		resolvedBound, err := amm.recursivelyResolveUp(parent, app.Meta.Name, bound, usePermTree, "route")
```

Comment:
Replace resourceType string for int, and use a constant named route with a set integer value 
and another constant named channel